File: ui/composite/TrackPanelArea.py
__author__ = 'toramisu'
from ui import *
from .TrackPanel import TrackPanel


class TrackPanelArea(QWidget):
    def __init__(self, parent):
        super(TrackPanelArea, self).__init__(parent)
        self.trackPanelStack = QWidget(self)
        self.trackPanelStack.resize(TIMELINE_TRACK_PANEL_DEF_WIDTH, 10)
        self.vbox = QVBoxLayout(self.trackPanelStack)
        self.vbox.setSpacing(0)
        self.trackPanels = []
        # self.setWidget(self.trackPanelStack)
        # self.show()

    def addTrackPanel(self, trackInfo):
        tp = TrackPanel()
        tp.setTrackName(trackInfo.name)
        self.trackPanelStack.resize(self.trackPanelStack.width(), self.trackPanelStack.height() + tp.height())
        self.vbox.addWidget(tp)
        self.trackPanels.append(tp)
        # self.scrollContentsBy(500,10)
        print(self.height())
        pass

File: view/timeline/trackPanel/TrackPanelArea.cpp
//
// Created by manoyuria on 2015/6/18.
//
#include "TrackPanelArea.h"
#include "TrackPanel.h"

TrackPanelArea::TrackPanelArea(QWidget *parent) : QWidget(parent) {
    QWidget *stack = new QWidget(this);
    stack->resize(TIMELINE_TRACK_PANEL_DEF_WIDTH, 10);
    vbox = new QVBoxLayout(stack);
    vbox->setSpacing(0);
    vbox->setContentsMargins(0, 0, 0, 0);
}

void TrackPanelArea::add(TrackInfo *trackInfo) {
    TrackPanel *tp = new TrackPanel();
    tp->setTrackName(trackInfo->name);
    _setHeight(vbox->parentWidget(), vbox->parentWidget()->height() + tp->height());
    vbox->addWidget(tp);
    qDebug() << this << this->height();
//    tp = TrackPanel()
//    tp.setTrackName(trackInfo.name)
//    self.trackPanelStack.resize(self.trackPanelStack.width(), self.trackPanelStack.height() + tp.height())
//    self.vbox.addWidget(tp)
//    self.trackPanels.append(tp)
//# self.scrollContentsBy(500,10)
//    print(self.height())
}
